src/video/x11/SDL_x11video.c

```diff
@@ -460,6 +460,14 @@ X11_VideoInit(_THIS)
     X11_InitXfixes(_this);
 #endif /* SDL_VIDEO_DRIVER_X11_XFIXES */
 
+#ifndef X_HAVE_UTF8_STRING
+    SDL_LogError(SDL_LOG_CATEGORY_VIDEO, "X server doesn't support UTF8_STRING, a feature introduced in 2000! This is likely to become a hard error in a future libSDL2.");
+#endif
+
+    if (X11_XSupportsLocale() != True) {
+        return SDL_SetError("Current locale not supported by X server, cannot continue.");
```

Comment:
We shouldn't introduce a video initialization failure here, even if it seems like it should always succeed. Instead we should check this when we're setting the title, and fail to do so there if necessary.

Reply:
Oh, all right. That makes sense, I appreciate your efforts in writing fault tolerant software. I write a lot of Rust and a lot of those folks value correctness above all else. (Meaning, crash/refuse to continue if anything is wrong.) Glad to know I don't have to code that way here, should we work together more beyond this patch. Fixed in 513e0eb.